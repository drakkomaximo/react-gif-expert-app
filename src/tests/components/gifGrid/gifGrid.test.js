import { shallow } from "enzyme"
import GifGrid from "../../../components/gifGrid"
import { useFetchGifs } from "../../../hooks/useFetchGifs"
import '@testing-library/jest-dom'
jest.mock("../../../hooks/useFetchGifs")

describe('Prueba de <GifGrid />', () => {
    
    const category = 'One Punch'
    /* let wrapper = shallow( <GifGrid category={ category }/>) */

    test('debe funcionar correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar items cuando se cargan imagenes con el useFetchGifs', () => {

        const gifs = [{
            id:'ABC',
            url: 'https://localhskafdgjkashgdkjh',
            title:'Prueba'
        },
        {
            id:'ABC4u',
            url: 'https://localhskafdgjkashgdkjh',
            title:'Prueba'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe( true )
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length )
    })
})
