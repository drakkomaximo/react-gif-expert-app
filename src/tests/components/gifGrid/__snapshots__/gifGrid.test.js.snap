// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifGrid /> debe funcionar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba de <GifGrid /> debe mostrar items cuando se cargan imagenes con el useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Prueba"
      url="https://localhskafdgjkashgdkjh"
    />
    <GifGridItem
      id="ABC4u"
      key="ABC4u"
      title="Prueba"
      url="https://localhskafdgjkashgdkjh"
    />
  </div>
</Fragment>
`;
